modules/lm-coursier/src/main/scala/lmcoursier/CoursierDependencyResolution.scala

```diff
@@ -122,7 +122,8 @@ class CoursierDependencyResolution(conf: CoursierConfiguration) extends Dependen
       .map {
         case (config, dep) =>
           val dep0 = dep.withExclusions(dep.exclusions ++ excludeDependencies)
-          (ToCoursier.configuration(config), ToCoursier.dependency(dep0))
+          val dependency = ToCoursier.dependency(dep0)
+          (ToCoursier.configuration(config), dependency.withOptional(dependency.optional || configuration.missingOk))
```

Comment:
I feel like asking for optional dependencies has slightly different meaning to asking for the whole graph while allowing final error to be an error - https://github.com/sbt/librarymanagement/blob/v1.3.1/ivy/src/main/scala/sbt/internal/librarymanagement/IvyActions.scala#L323-L340. For example, you can ask for dependencies A whose dependency B is missing from Maven Central. We'd still want missingOk to recover from it.

Reply:
it does recover from that, since all transitive deps are marked as optional too